backend/src/handlers.ts

```diff
@@ -0,0 +1,183 @@
+import {Request, Response} from "express";
+import {LAMPORTS_PER_SOL, ParsedInstruction, PartiallyDecodedInstruction, PublicKey} from "@solana/web3.js";
+import {findGatewayPass, GatekeeperService} from "@identity.com/gateway-solana-client";
+import Evervault from "@evervault/sdk";
+import {TOKEN_PROGRAM_ID} from "@solana/spl-token";
+import {
+  connection,
+  storage
+} from "./index";
+import {
+  FUNDER_ATA,
+  FUNDER_AUTHORITY,
+  GATEKEEPER,
+  GATEKEEPER_ATA,
+  GATEKEEPER_AUTHORITY,
+  MINT,
+  NETWORK,
+  NETWORK_ATA
+} from "./config";
+import {Wallet} from "@project-serum/anchor";
+
+const handleDocumentUpload = async (request: Request, _: Response) => {
+  console.log("Encrypting and uploading documents")
+  const documentUuid = request.body.event.data.uuid;
+
+  const evervault = new Evervault(process.env.EVERVAULT_API_KEY);
+
+  const result = await evervault.run('socure-poc-cage', {
+    documentUuid
+  });
+
+  await storage.store(request.body.event.customerUserId, 'image-data.json',
+    JSON.stringify({
+      iv: result.result.iv,
+      key: result.result.key
+    })
+  );
+
+  await storage.store(request.body.event.customerUserId, 'image.zip.enc', Buffer.from(result.result.data.data));
+}
+
+const handleVerificationComplete = async (request: Request, _response: Response) => {
+  console.log("Uploading PII and issuing token");
+  if (request.body.event.data.documentVerification.decision.value !== 'accept') {
+    console.log("Validation failed");
+    return;
+  }
+
+  const gatekeeperService = await GatekeeperService.build(
+    NETWORK,
+    GATEKEEPER,
+    {
+      clusterType: 'devnet',
+      wallet: new Wallet(GATEKEEPER_AUTHORITY),
+    }
+  );
+
+  const subject = new PublicKey(request.body.event.customerUserId);
+  const passAccount = await GatekeeperService.createPassAddress(
+    subject,
+    NETWORK
+  );
+
+  // Store PII
+  // await storage.store(passAddress.toBase58(), 'pii.json', JSON.stringify(request.body, null, 2));
```

Comment:
We need to activate this... I'll create a ticket... I disabled it in Lisbon for our challenge

Reply:
Thanks!